api/pkg/apis/v1alpha1/managers/jobs/jobs-manager.go

```diff
@@ -47,6 +51,41 @@ func (s *JobsManager) Init(context *contexts.VendorContext, config managers.Mana
 	} else {
 		return err
 	}
+
+	baseUrl, err := utils.GetString(s.Manager.Config.Properties, "baseUrl")
+	if err != nil {
+		return err
+	}
+
+	s.interval = utils.ReadInt32(s.Manager.Config.Properties, "interval", 0)
+
+	clientOptions := make([]utils.ApiClientOption, 0)
+
+	if caCert, ok := os.LookupEnv(constants.ApiCertEnvName); ok {
+		clientOptions = append(clientOptions, utils.WithCertAuth(caCert))
+	}
+
+	if utils.ShouldUseSATokens() {
+		clientOptions = append(clientOptions, utils.WithServiceAccountToken())
+	} else {
+		user, err := utils.GetString(s.Manager.Config.Properties, "user")
+		if err != nil {
+			return err
+		}
+
+		password, err := utils.GetString(s.Manager.Config.Properties, "password")
+		if err != nil {
+			return err
+		}
+		clientOptions = append(clientOptions, utils.WithUserPassword(user, password))
+	}
+
+	client, err := utils.NewAPIClient(context.Background(), baseUrl, clientOptions...)
```

Comment:
context.Background only works for pollObjects(). Other places using ApiClient need to use context in the function parameter.

Reply:
You are right. In this PR, this is the only place of using NewAPIClient in symphony-api. Eventually, we will fix all api calls but we don't want the PR to be very big so this replacement will be completed in another task by @iwangjintian. Ideally, when every time creating the apiclient, we need to use its inherited context or context.TODO().

In the upcoming PR which includes K8S changes, you will see apiClient in symphony-k8s will use its own context.